File: frontend/src/api/chatHistory.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request'

/** List All Chat History By Page For Admin POST /api/chatHistory/admin/list/page/vo */
export async function listAllChatHistoryByPageForAdminApiChatHistoryAdminListPageVoPost(
  body: Record<string, any>,
  options?: { [key: string]: any }
) {
  return request<any>('/api/chatHistory/admin/list/page/vo', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** List App Chat History GET /api/chatHistory/app/${param0} */
export async function listAppChatHistoryApiChatHistoryAppAppIdGet(
  params: API.listAppChatHistoryParams,
  options?: { [key: string]: any }
) {
  const { appId, sessionId, ...queryParams } = params
  return request<any>(`/api/chatHistory/app/${appId}`, {
    method: 'GET',
    params: {
      sessionId: sessionId ?? '',
      ...queryParams,
    },
    ...(options || {}),
  })
}

File: backend/api-gateway/api/chat_history.py
"""Chat history routes forwarded by gateway to app-service."""

from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Header, Query

from proxy.app_proxy import AppProxy


router = APIRouter(prefix="/chatHistory", tags=["chatHistory"])


@router.get("/app/{app_id}")
async def list_app_chat_history(
    app_id: int,
    session_id: str = Query(default="", alias="sessionId"),
    authorization: str | None = Header(default=None, alias="Authorization"),
):
    params: dict[str, Any] = {"session_id": session_id}

    proxy = AppProxy()
    return await proxy.request_json(
        method="GET",
        path=f"/api/chatHistory/app/{app_id}",
        authorization=authorization,
        params=params,
    )


@router.post("/admin/list/page/vo")
async def list_all_chat_history_by_page_for_admin(
    payload: dict[str, Any],
    authorization: str | None = Header(default=None, alias="Authorization"),
):
    proxy = AppProxy()
    return await proxy.request_json(
        method="POST",
        path="/api/chatHistory/admin/list/page/vo",
        authorization=authorization,
        json_body=payload,
    )
